astropy/coordinates/sites.py

```diff
@@ -0,0 +1,134 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Observatories accessible without internet access originate from the IRAF
+Observatory Database, and are stored in ``data/observatories.json``.  This is
+inteded mainly as a fallback file, and the online file is where new changes
+should go.
+
+Additions or corrections to the observatory list can be submitted via Pull
+Request to the [astropy-data GitHub repository](https://github.com/astropy/astropy-data),
+updating the ``location.json`` file.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import json
+from difflib import get_close_matches
+from collections import Mapping
+
+from ..utils.data import get_pkg_data_contents, get_file_contents
+from .earth import EarthLocation
+from .. import units as u
+
+
+class SiteRegistry(Mapping):
+    """
+    A bare-bones registry of EarthLocation objects.
+
+    This acts as a mapping (dict-like object) but with the important caveat that
+    it's always transforms its inputs to lower-case.  So keys are always all
+    lower-case, and even if you ask for something that's got mixed case, it will
+    be interepreted as the all lower-case version.
+    """
+    def __init__(self):
+        # the keys to this are always lower-case
+        self._lowercase_names_to_locations = {}
+        # these can be whatever case is appropriate
+        self._names = []
+
+    def __getitem__(self, site_name):
+        """
+        Returns an EarthLocation for a known site in this registry.
+
+        Parameters
+        ----------
+        site_name : str
+            Name of the observatory (case-insensitive).
+
+        Returns
+        -------
+        site : `~astropy.coordinates.EarthLocation`
+            The location of the observatory.
+        """
+        if site_name.lower() not in self._lowercase_names_to_locations:
+            # If site name not found, find close matches and suggest them in error
+            close_names = get_close_matches(site_name, self._lowercase_names_to_locations)
+            close_names = sorted(close_names, key=lambda x: len(x))
+            if close_names:
+                errmsg = ('Site not in database. Use ``get_names()`` '
```

Comment:
Sorry to come to this only now, and so trivial too, but shouldn't the error messages be identical except for the addition of suggested names? I.e., here, shouldn't you list what name was put in, and below, also give the hint of using `get_names()` to see the full list? If so, just make a standard message and append "Did you mean one of ..." for the case of close names.